### PR TITLE
Test for distributed RL with RPC

### PR DESCRIPTION
**Pull request by jbschlosser**
Addresses one item in #46321

## Background
This is a test version of the RL RPC example defined [here](https://github.com/pytorch/examples/blob/master/distributed/rpc/rl/main.py) and [here](https://pytorch.org/tutorials/intermediate/rpc_tutorial.html), with the following differences:
* It defines and uses a `DummyEnv` to avoid a dependency on `gym`. The `DummyEnv` simply returns random states & rewards for a small number of iterations.
* It removes the `ArgumentParser` and utilizes `RpcAgentTestFixture` + hard-coded constants for configuration and launching.
* It changes the worker names to match what the internal Thrift RPC tests expect.

The code is purposefully kept very similar to the original example code outside of these differences.

## Test Plan
```
pytest test/distributed/rpc/test_tensorpipe_agent.py -k test_rl_rpc -vs
pytest test/distributed/rpc/test_process_group_agent.py -k test_rl_rpc -vs
```

### CODE REVIEW

File: torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torch.distributed.rpc import RRef, rpc_sync, rpc_async, remote
+from torch.distributions import Categorical
+
+from torch.testing._internal.dist_utils import dist_init
+from torch.testing._internal.distributed.rpc.rpc_agent_test_fixture import RpcAgentTestFixture
+
+TOTAL_EPISODE_STEP = 5000
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py:19]: how much time does this test take? Do we need to use a smaller step to speed up testing?

File: torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+class DummyEnv:
+    r"""
+    A dummy environment that implements the required subset of the OpenAI gym
+    interface. It exists only to avoid a dependency on gym for running the
+    tests in this file. It is designed to run for a set max number of iterations,
+    returning random states and rewards at each step.
+    """
+    def __init__(self, state_dim=4, action_dim=2, num_iters=10, reward_threshold=475.0):
+        self.state_dim = state_dim
+        self.action_dim = action_dim
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py:73]: this is not used?

**Review by mrshenli — APPROVED** [review]
LGTM. 

Quantization test failures no longer exist in master, could you please rebase to clear CI signals?

Question, is there anything in this test that we can assert on?

File: torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py (hunk 2 above)
Comment (jbschlosser) [reply torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py:73]: Ah you're right- I realized after adding this that the action dim is hard-coded in the Policy above. Will remove

File: torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py (hunk 1 above)
Comment (jbschlosser) [reply torch/testing/_internal/distributed/rpc/examples/reinforcement_learning_rpc_test.py:19]: Test takes ~2sec, which is on the same order as, for example, `test_batch_updating_parameter_server`. In practice, I don't think it ever hits this step number, since the accumulated reward will pass the reward threshold relatively quickly.

**Comment by jbschlosser** [comment]
> Question, is there anything in this test that we can assert on?

Hm good question- there are no asserts in the other example test either (`test_batch_updating_parameter_server`). Since we only care about checking the API, I don't think we want to assert anything related to the RL task (e.g. checking reward, etc.).

**Comment by jbschlosser** [comment]
The internal FB tests are failing because they set the default tensor type to double: [tensorpipe tests](https://www.internalfb.com/intern/diffusion/FBS/browsefile/master/fbcode/caffe2/test/distributed/rpc/fb/test_tensorpipe_agent.py?lines=16) / [process group tests](https://www.internalfb.com/intern/diffusion/FBS/browsefile/master/fbcode/caffe2/test/distributed/rpc/fb/test_process_group_agent.py?lines=16). Currently trying to track down why this is done.
